### DISCUSSION TITLE
what about multiple tcp connection between nats-server？

### DISCUSSION THREAD

**Discussion by song0071000** [nats-io/nats-server#6193] [Ideas]
We are experiencing issues with our business requests that span across clusters, which are affected by TCP head-of-line blocking problems.

Does the NATS function depend on the order of messages?

What happens if we establish multiple TCP connections between NATS servers?

any ideas？

**Comment by neilalexander** [comment]
When you say "across clusters", do you mean a gateway/supercluster-style configuration, or do you mean route connections within a cluster?

If the latter then you want https://docs.nats.io/running-a-nats-service/configuration/clustering/v2_routes connection pooling.

**Reply by song0071000** [reply]
Thanks for reply!
I mean supercluster scenario, the connection between gateways.

**Reply by AlexeiSosnov** [reply]
You can also establish multiple connections (e.g. between central node cluster - leaf nodes) by using multiple accounts, and use it to "shard" traffic between nodes (e.g. traffic of high importance commands goes through one connection, and traffic formed from large messages goes through another connection)

**Reply by song0071000** [reply]
> You can also establish multiple connections (e.g. between central node cluster - leaf nodes) by using multiple accounts, and use it to "shard" traffic between nodes (e.g. traffic of high importance commands goes through one connection, and traffic formed from large messages goes through another connection)

That's a great idea! 
However, I'm curious whether multiple accounts can be used for the gateway connection.
